cow_instrument/InstrumentTree.h

```diff
@@ -17,30 +16,38 @@ class Detector;
  */
 class InstrumentTree {
 public:
-  InstrumentTree(std::unique_ptr<const Node> &&root, size_t nDetectors);
-
-  InstrumentTree(const InstrumentTree& other);
-
-  std::unique_ptr<NodeIterator> iterator() const;
+  InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors);
+  InstrumentTree(const InstrumentTree& original) = default;
 
   const Node &root() const;
 
   const Detector &getDetector(size_t detectorIndex) const;
 
   unsigned int version() const;
 
-  std::unique_ptr<const InstrumentTree> modify(const Command &command) const;
+  InstrumentTree modify(size_t node, const Command &command) const;
+  InstrumentTree modify(const Node *node, const Command &command) const;
```

Comment:
Would it make sense to have something like
```cpp
class NodeIndex {
private:
  size_t index;
  // or
  // const Node *node;
};
```
This index object could be used for all interactions with the instrument tree?